src/app/SERVICE/tasks.service.ts

```diff
@@ -1,61 +1,77 @@
 import { Injectable } from '@angular/core';
-import { Task } from '../MODELS/task.model';
-import { BehaviorSubject } from 'rxjs';
+import { Task } from '../models/task.model';
+import { BehaviorSubject, map } from 'rxjs';
+import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
+import { Router } from '@angular/router';
+import { response } from 'express';
 
 @Injectable({
   providedIn: 'root'
 })
 export class TasksService {
+  updatedTaskList: any;
 
-  constructor() {
+  // readonly ROOT_URL = "localhost:3000" // ATM proxy handles this root
+  tasksApiUrl: string = "http://localhost:3000/tasks/"
+
+  constructor(private http: HttpClient, private router: Router) {
   }
-  private tasks = new BehaviorSubject<Task[]>([
-    {
-      id: "8c410ecc-a204-43b6-9582-1fee127f858a",
-      title: "Grocery Shopping",
-      description: "Buy milk, bread, eggs, and fruits",
-      type: "Personal",
-      createdOn: new Date(),
-      status: "To Do",
-    },
-    {
-      id: "d468c77f-844c-4af6-b590-af2d8c7a2b9d",
-      title: "Finish Work Report",
-      description: "Complete the Q2 sales report for the manager",
-      type: "Work",
-      createdOn: new Date(2024, 6, 15),
-      status: "In Progress",
-    },
-    {
-      id: "e70626f2-4baf-4c3e-94ca-b0a9ce4c7a39",
-      title: "Pay Bills",
-      description: "Pay electricity and internet bills",
-      type: "Personal",
-      createdOn: new Date(2024, 6, 12),
-      status: "In Progress",
-    },
-  ])
+  private tasks = new BehaviorSubject<Task[]>([])
 
   availableTasks$ = this.tasks.asObservable()
 
+  async getAllTasks() {
```

Comment:
you dont need to specify that its an async method also await doesnt need to be specified either

Reply:
in general in angular you would just return the observable from the method and then you can use that observable however you like, if you want to update the tasks subject you can do that from the components, same applies to all other methods